source/examples/demo-stages/CMakeLists.txt

```diff
@@ -0,0 +1,261 @@
+    ${source_path}/LightTestStage.cpp
+    ${source_path}/LightTestPipeline.cpp
+)
+
+# Group source files
+set(header_group "Header Files (API)")
+set(source_group "Source Files")
+source_group_by_path(${include_path} "\\\\.h$|\\\\.inl$"
+    ${header_group} ${headers})
+source_group_by_path(${source_path}  "\\\\.cpp$|\\\\.c$|\\\\.h$|\\\\.inl$"
+    ${source_group} ${sources})
+
+
+#
+# Create library
+#
+
+# Build library
+add_library(${target}
+    ${sources}
+    ${headers}
+)
+
+# Create namespaced alias
+add_library(${META_PROJECT_NAME}::${target} ALIAS ${target})
+
+# Export library for downstream projects
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+
+# Check for availability of module; use pre-generated version if not found
+if (WriterCompilerDetectionHeaderFound)
+    write_compiler_detection_header(
+        FILE ${feature_file}
+        PREFIX ${target_id}
+        COMPILERS AppleClang Clang GNU MSVC
+        FEATURES cxx_noexcept
+        VERSION 3.2
+    )
+else()
+    file(
+        COPY ${PROJECT_SOURCE_DIR}/source/codegeneration/${target}_features.h
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/include/${target}
+        USE_SOURCE_PERMISSIONS
+    )
+endif()
+
+# Create API export header
+generate_export_header(${target}
+    EXPORT_FILE_NAME  ${export_file}
+    EXPORT_MACRO_NAME ${export_macro}
+)
+
+
+#
+# Project options
+#
+
+set_target_properties(${target}
+    PROPERTIES
+    ${DEFAULT_PROJECT_OPTIONS}
+    FOLDER "${IDE_FOLDER}"
+)
+
+
+#
+# Include directories
+#
+
+target_include_directories(${target}
+    PRIVATE
+    ${DEFAULT_INCLUDE_DIRECTORIES}
+    ${PROJECT_BINARY_DIR}/source/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    ${CMAKE_CURRENT_BINARY_DIR}/include
+    ${GLM_INCLUDE_DIR}
+
+    PUBLIC
+
+    INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/source/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+
+#
+# Libraries
+#
+
+target_link_libraries(${target}
+    PRIVATE
+    ${OPENGL_LIBRARIES}
+    ${META_PROJECT_NAME}::gloperate
+    cpplocate::cpplocate
+    cppexpose::cppexpose
+    cppassist::cppassist
+    glbinding::glbinding
+    globjects::globjects
+
+    PUBLIC
+    ${DEFAULT_LIBRARIES}
+
+    INTERFACE
+)
+
+
+#
+# Compile definitions
+#
+
+target_compile_definitions(${target}
+    PRIVATE
+
+    PUBLIC
+    $<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:${target_id}_STATIC_DEFINE>
+    ${DEFAULT_COMPILE_DEFINITIONS}
+    GLM_FORCE_RADIANS
+
+    INTERFACE
+)
+
+
+#
+# Compile options
+#
+
+target_compile_options(${target}
+    PRIVATE
+
+    PUBLIC
+    ${DEFAULT_COMPILE_OPTIONS}
+
+    INTERFACE
+)
+
+
+#
+# Linker options
+#
+
+target_link_libraries(${target}
+    PRIVATE
+
+    PUBLIC
+    ${DEFAULT_LINKER_OPTIONS}
+
+    INTERFACE
+)
+
+
+#
+# Deployment
```

Comment:
I think the deployment section can be copied from any other example. Only the plugin/shared library has to get installed. (e.g., https://github.com/cginternals/gloperate/blob/master/source/examples/gloperate-text-example/CMakeLists.txt#L150)